File: rust/src/models/price_component.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{fee::Fee, price_component_id::PriceComponentId, product_id::ProductId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PriceComponent {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub fee: Option<Fee>,

    pub id: PriceComponentId,

    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub product_id: Option<ProductId>,
}

impl PriceComponent {
    pub fn new(id: PriceComponentId, name: String) -> Self {
        Self {
            fee: None,
            id,
            name,
            product_id: None,
        }
    }
}

File: typescript/src/models/priceComponent.ts
// this file is @generated
import { type Fee, FeeSerializer } from "./fee";
import { type PriceComponentId, PriceComponentIdSerializer } from "./priceComponentId";
import { type ProductId, ProductIdSerializer } from "./productId";

export interface PriceComponent {
  fee?: Fee | null;

  id: PriceComponentId;

  name: string;

  productId?: ProductId | null;
}

export const PriceComponentSerializer = {
  _fromJsonObject(object: any): PriceComponent {
    return {
      fee:
        object["fee"] != null ? FeeSerializer._fromJsonObject(object["fee"]) : undefined,
      id: PriceComponentIdSerializer._fromJsonObject(object["id"]),
      name: object["name"],
      productId:
        object["product_id"] != null
          ? ProductIdSerializer._fromJsonObject(object["product_id"])
          : undefined,
    };
  },

  _toJsonObject(self: PriceComponent): any {
    return {
      fee: self.fee != null ? FeeSerializer._toJsonObject(self.fee) : undefined,
      id: PriceComponentIdSerializer._toJsonObject(self.id),
      name: self.name,
      product_id:
        self.productId != null
          ? ProductIdSerializer._toJsonObject(self.productId)
          : undefined,
    };
  },
};
